File: Beecrowd-Problems/C++/done!/problem1013.cpp
#include <iostream>

int maiorAB(int a, int b);

int main(){

    int a, b, c;
    int result;

    std::cin >> a;
    std::cin >> b;
    std::cin >> c;

    result = maiorAB(a, b);
    result = maiorAB(result, c);

    std::cout << result << " eh o maior\n";

    return 0;
}

int maiorAB(int a, int b){
    int maior = (a + b + abs(a - b)) / 2;
    if (maior == a)
        return a;
    return b;
}

File: Beecrowd-Problems/C#/done!/problem1013.cs
using System;
using System.Globalization;

namespace Program {
    class Program {

        static Tuple<int, int, int> ReadAndParseInput(){
            string line = Console.ReadLine();
            var values = line.Split(' ');
            int a = int.Parse(values[0]);
            int b = int.Parse(values[1]);
            int c = int.Parse(values[2]);
            return Tuple.Create(a, b, c);
        }

        static int MaiorAB(int a, int b) {
            return (a + b + Math.Abs(a - b)) / 2;
        }

        static void Main(string[] args) {

            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
            CultureInfo.CurrentUICulture = CultureInfo.InvariantCulture;

            var inputTuple = ReadAndParseInput();
            int a = inputTuple.Item1;
            int b = inputTuple.Item2;
            int c = inputTuple.Item3;

            int maior = MaiorAB(a, b);
            maior = MaiorAB(maior, c);

            Console.WriteLine($"{maior} eh o maior");
        }
    }
}